packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -672,8 +677,10 @@ class PopupMenuButton<T> extends StatefulWidget {
     this.tooltip: 'Show menu',
     this.elevation: 8.0,
     this.padding: const EdgeInsets.all(8.0),
-    this.child
+    this.child,
+    this.icon,
   }) : assert(itemBuilder != null),
+       assert(child == null || icon == null), // or both
```

Comment:
you're hitting an analyzer bug: https://github.com/dart-lang/sdk/issues/30288
change the `||` to `&&`, the `==` to `!=`, and prefix the whole thing with a `!` and it'll mean the same thing but work.
`assert(!(child != null && icon != null))`